### PR TITLE
KFSPTS-16227 save PVEN doc when there are business rules errors

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/sys/identity/ExcludeSubmitterSeparationOfDutiesRoleTypeService.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        return membershipInfos;
+    }
+
+    private String getSubmitter(String documentId) {
+        String submitterPrincipalId = null;
+
+        String principalId = workflowDocumentService.getRoutedByPrincipalIdByDocumentId(documentId);
+        List<ActionTaken> actionTakenDTOs = workflowDocumentService.getActionsTaken(documentId);
+        for (ActionTaken actionTaken : actionTakenDTOs) {
+            if (principalId.equals(actionTaken.getPrincipalId())) {
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/identity/ExcludeSubmitterSeparationOfDutiesRoleTypeService.java:36]: I would suggest doing this principalId comparison in a null-safe way.

File: src/main/resources/edu/cornell/kfs/vnd/document/workflow/CuVendorMaintenanceDocument.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          </split>
+        </routePath>
+      </routePaths>
+      <routeNodes>
+        <start name="AdHoc" />
+        <split name="RequiresApproval">
+          <type>org.kuali.kfs.sys.document.workflow.SimpleBooleanSplitNode</type>
+        </split>
+        <role name="Submitter">
+          <!--<ruleTemplate>KualiVendorDocumentTypeTemplate</ruleTemplate>-->
```
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/vnd/document/workflow/CuVendorMaintenanceDocument.xml:34]: Can these commented-out elements be removed?

File: src/main/java/edu/cornell/kfs/sys/identity/ExcludeSubmitterSeparationOfDutiesRoleTypeService.java (hunk 1 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/sys/identity/ExcludeSubmitterSeparationOfDutiesRoleTypeService.java:36]: Updating

File: src/main/resources/edu/cornell/kfs/vnd/document/workflow/CuVendorMaintenanceDocument.xml (hunk 2 above)
Comment (db483) [reply src/main/resources/edu/cornell/kfs/vnd/document/workflow/CuVendorMaintenanceDocument.xml:34]: These were in the exported xml but I can remove them.

File: src/main/java/edu/cornell/kfs/sys/identity/ExcludeSubmitterSeparationOfDutiesRoleTypeService.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package edu.cornell.kfs.sys.identity;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.codec.binary.StringUtils;
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/identity/ExcludeSubmitterSeparationOfDutiesRoleTypeService.java:6]: I would suggest using the Commons Lang3 version of StringUtils instead, for consistency with the rest of our code.

Comment (db483) [reply src/main/java/edu/cornell/kfs/sys/identity/ExcludeSubmitterSeparationOfDutiesRoleTypeService.java]: good point, not sure how that got imported. Updating

**Comment by db483** [comment]
I made all the changes, this can be reviewed again. Thank you.

**Pull request merged**